arctic/serialization/numpy_arrays.py

```diff
@@ -218,16 +222,19 @@ def deserialize(self, data, columns=None):
             if index:
                 columns = columns[:]
                 columns.extend(meta[INDEX])
-            if len(columns) > len(set(columns)):
-                raise Exception("Duplicate columns specified, cannot de-serialize")
+                columns = list(set(columns))
```

Comment:
I'm not sure I see this as a win. It seems like the caller may have a bug if they are specifying duplicate columns, we're just hiding the error now

Reply:
The current logic is confusing when subsetting data frames with indexes. For example, if you have the data frame:
index: date, security
columns: price, volume

The logic works if the user passes:
['price']
Raises a duplicate columns error when passing:
['date','security','price']

I don't see the value in the check - it should just do the right thing..

Reply:
Using pandas nomenclature the columns and the index are separate. If there is an index, you always get it back, even if you specify a subset of columns (and even if they do not include the index columns). Maybe the documentation should be improved. If for example, you specify price and security, you'll still get date as well as price and security, so your fix would only introduce more weirdness (in my opinion).

Reply:
We could remove index columns from columns and then check for duplicates. This keeps the nomenclature but keeps the user interface 'minimum surprise'. Or raise an error saying they have included index columns in the column list.

Reply:
The result would be the same though, no? You'd supply index columns and it wont complain. I foresee someone opening a bug complaining they only specified 1 of 3 index columns but still got all 3 back.

Reply:
@TomTaylorLondon not trying to hold up the PR. Is allowing the user to specify all columns (even index columns) something you need? Could emit a warning log and allow them to use the index columns. Would that be ok?

Reply:
Yes - it's bitten us a few times. As a user as your never quite sure what is in the index and what is in the columns, so it's hard to know exactly what to specify here.

Reply:
ok, so lets leave your change in then. I'll probably tweak `get_info` to separate out the index labels from the columns so then you will be able to know. Whenever i get that done, I'll probably change this back to columns only (no index labels). Sound ok?

Reply:
Sure

Reply:
Actually, in retrospect, that means breaking the API for clients. How about we keep the fuzziness for clients and simply output a warning instead?

Reply:
ok, lets see a warning :) but i still think that get info should change, otherwise how would you ever know how to rid yourself of the warning?

Reply:
Agree with get_info change

Reply:
so sounds like you just need to fix the broken tests and add the log and we're all set :D